analyzer/compliance-analyzer/src/main/java/eu/fasten/analyzer/complianceanalyzer/ComplianceAnalyzerPlugin.java

```diff
@@ -0,0 +1,278 @@
+
+    /**
+     * How will the patched Job file be called.
+     */
+    private static final String PATCHED_JOB_FILE_NAME = "patchedJob.yaml";
+
+    public ComplianceAnalyzerPlugin(PluginWrapper wrapper) {
+        super(wrapper);
+    }
+
+    @Extension
+    public static class CompliancePluginExtension implements KafkaPlugin, DBConnector {
+        private String consumerTopic = "fasten.RepoCloner.out";
+        private static DSLContext dslContext;
+        private boolean processedRecord = false;
+        private Throwable pluginError = null;
+        private final Logger logger = LoggerFactory.getLogger(CompliancePluginExtension.class.getName());
+        private static RocksDao rocksDao;
+        private String outputPath;
+
+        private String artifactID;
+        private String repoUrl;
+        private String repoPath;
+
+
+        public void setRocksDao(RocksDao rocksDao) {
+            CompliancePluginExtension.rocksDao = rocksDao;
+        }
+
+         @Override
+         public void setDBConnection(DSLContext dslContext) {
+            CompliancePluginExtension.dslContext = dslContext;
+         }
+
+
+        @Override
+        public Optional<List<String>> consumeTopic() {
+            return Optional.of(Collections.singletonList(consumerTopic));
+        }
+
+        @Override
+        public void setTopic(String topicName) {
+            this.consumerTopic = topicName;
+        }
+
+        @Override
+        public void consume(String record) {
+            this.processedRecord = false;
+            this.pluginError = null;
+            var consumedJson = new JSONObject(record);
+            if (consumedJson.has("input")) {
+                consumedJson = consumedJson.getJSONObject("input");
+            }
+            this.repoUrl = consumedJson.getString("repoUrl");
+            var consumedJsonPayload = new JSONObject(record);
+            if (consumedJsonPayload.has("payload")) {
+                consumedJsonPayload = consumedJsonPayload.getJSONObject("payload");
+            }
+            this.repoPath = consumedJsonPayload.getString("repoPath");
+            this.artifactID = consumedJsonPayload.getString("artifactId");
+
+            logger.info("Repo url: " + repoUrl);
+            logger.info("Path to the cloned repo: " + repoPath);
+            logger.info("Artifact id: " + artifactID);
+
+            if (repoUrl == null) {
+                logger.info("No available URL for the specified project.\n");
+                logger.info("Skipping analysis....");
+                return;
+            }
+
+            /**
+             * Connect to Cluster and
+             * start a Kubernetes Job
+             */
+
+            ConnectToCluster();
+            CreateJobConfig();
+            ApplyK8sJob();
+
+            processedRecord = true;
+        }
+
+        public void ConnectToCluster() {
+            // Google Cloud credentials
+            String jsonPath = "./fasten-ca08747cdc4f.json";
```

Comment:
> 1. Are those real credentials and if so, do we really want to publish a private key?!

I absolutely agree. Those shouldn't have been pushed. I'll get rid of this file with [BFG](https://rtyley.github.io/bfg-repo-cleaner), make it an argument `--cluster-credentials` and invalidate the current credentials.

> 2. It only works if run from the root of the plugin. In all other cases it cannot find this file and hence fails. It's better not to have it hardcoded but supplied through environmental variable or an argument.

Agreed. Given that I will get rid of that file, we would face the same problem with our [`docker`](https://github.com/fasten-project/fasten/tree/endocode/compliancePlugin/analyzer/compliance-analyzer/docker) folder, as our plugin solely relies on our Kubernetes objects to be able to run. I've read a bit about using relative paths in Maven projects, and I found [this](https://maven.apache.org/plugin-developers/common-bugs.html#resolving-relative-paths). Long story short, they recommend doing the following:

> In order to guarantee reliable builds, Maven and its plugins must manually resolve relative paths against the project's base directory. A simple idiom like the following will do just fine:
> 
> ```java
> File file = new File( path );
> if ( !file.isAbsolute() )
> {
>     file = new File( project.getBasedir(), file );
> }
> ```

Is there a standard way to get a sort of `project.getBasedir()` in FASTEN?

Otherwise, I would try the Mojo parameters that are mentioned right after that quotation.

Reply:
There is no standard way to do it for us. We handle most (or maybe actually all) of those cases by proving absolute paths in env variables or arguments if possible.

Reply:
Ok. Providing our [`docker`](https://github.com/fasten-project/fasten/tree/endocode/compliancePlugin/analyzer/compliance-analyzer/docker) folder path as an argument might be cumbersome to the user as it is something used solely by our plugin.

Solved by using Maven resource files.
We will be able to run this plugin from anywhere with:
```bash
mvn exec:java -f path/to/pom.xml
```

Reply:
Done in https://github.com/fasten-project/fasten/commit/7975246450da52793713310dc6123c14c876daf5.